EIPS/eip-3386.md

```diff
@@ -0,0 +1,315 @@
+
+There exists only one of each ERC-721 token of with a given (address, id) pair. However, ERC-1155 tokens of a given (address, id) may have quantities greater than 1. Accordingly, the meaning of "Single" and "Batch" in each standard varies. In both standards, "single" refers to a single id, and "batch" refers to multiple ids. In ERC-1155, a single id event/function may involve multiple tokens, according to the `value` field.
+
+In building a common set of events and functions, we must be aware of these differences in implementation. The current implementation treats ERC-721 tokens as a special case where, in reference to the quantity of each {Base} token:
+- All parameters named `_amount`, MUST be `1`.
+- All parameters named `_amounts` MUST be either an empty list or a list of `1` with the same length as `_ids`.
+
+This keeps a consistent enumeration of tokens along with ERC-1155. Alternative implementations include:
+- A common interface with specialized functions. EX: `mintFromERC721`.
+- Separate interfaces for each type. EX: `ERC721Wrapper`, `ERC1155Wrapper`.
+
+#### ERC721, ERC1155
+The {Wrapper} MAY inherit from {ERC721} and/or {ERC1155} in order to call `super.mint`, directly. This is optional as minting {Base} tokens is not required in this standard. An "Initial NFT Offering" could use this to create a set of {Base} tokens within the contract, and directly distribute {Derivative} tokens.
+
+If the {Wrapper} does not inherit from {ERC721} or {ERC1155}, it MUST include calls to {IERC721} and {IERC1155} in order to transfer {Base} tokens.
+
+### Approval
+All of the underlying transfer methods are not tied to the {Wrapper}, but rather call the ERC-20/721/1155 transfer methods. Implementations of this standard MUST:
+- Either implement {Derivative} transfer approval for burning, and {Base} transfer approval for minting.
+- Or check for Approval outside of the {Wrapper} through {IERC721} / {IERC1155} before attempting to execute.
+
+## Backwards Compatibility
+Most existing implementations inherit from ERC-20, using functions `mint` and `burn`.
+Events:
+- Mint
+    - WK: DepositKittyAndMintToken
+    - NFTX: Mint
+
+- Burn
+    - WK: BurnTokenAndWithdrawKity
+    - NFTX: Redeem
+
+## Reference Implementation
+[ERC-3386 Reference Implementation](https://github.com/ashrowz/erc-3386)
+
+## Security Considerations
+Wrapper contracts are RECOMMENDED to inherit from burnable ERC-20 tokens. If they are not, the supply of the {Derivative} tokens MUST be controlled by the Wrapper. Similarly, price implementations MUST ensure that the supply of {Base} tokens is reflected by the {Derivative} tokens.
+
+With the functions `idBurn`, `idBurns`, users may target the most valuable NFT within the generic lot.  If there is a significant difference between tokens values of different ids, the contract SHOULD consider creating specialized pools (NFTX) or pricing (NFT20) to account for this.
+
+## References
```

Comment:
Please remove this section and hyperlink the text above with the references where applicable.